sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/Utility.java

```diff
@@ -350,6 +370,150 @@ public static Map<String, Object> parseNextLink(String nextLink) {
         return new HashMap<>();
     }
 
+    // Sentiment Analysis
+    public static Response<AnalyzeSentimentResultCollection> toAnalyzeSentimentResultCollectionResponse(
+        Response<SentimentResponse> response) {
+        return new SimpleResponse<>(response, toAnalyzeSentimentResultCollection(response.getValue()));
+    }
+
+    public static Response<AnalyzeSentimentResultCollection> toAnalyzeSentimentResultCollectionResponse2(
+        Response<AnalyzeTextTaskResult> response) {
+        return new SimpleResponse<>(response,
+            toAnalyzeSentimentResultCollection(((SentimentTaskResult) response.getValue()).getResults()));
```

Comment:
Do we have a way to check for the instance type/subtype before directly casting it to a subtype?

Reply:
I can add at a  line before but feel it is redundant since it is our implementation error it should not expect other things other than `SentimentTaskResult` here. If there is another class type, the runtime will throw `can't cost the class type`. Please correct me if I am wrong. I am not 100% sure about this.
```java
        if (!(response.getValue() instanceof SentimentTaskResult)) {
            throw new RuntimeException("Response value should be 'SentimentTaskResult' instance type");
        }
```

Reply:
I think throwing our own exception with a good error message should help. 
cc: @srnagar